### PR TITLE
Feat/ota 3677/docs links component

### PR DESCRIPTION
**Pull request by mhls**

### CODE REVIEW

File: ota-plus-web/app/reactapp-new/src/App.tsx [hunk 1]
```diff
@@ -1,15 +1,30 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
 import '../style/index.scss';
 
-class App extends React.PureComponent {
-  render() {
+const App: React.FC<{}> = () => {
+
+    const [data, setData] = useState('');
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp-new/src/App.tsx]: This is not needed in this ticket.

File: ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Row, Col } from 'antd';
+
+import './DocsLinks.scss';
+
+const DocsLinks: React.FC<{}> = () => {
+    return (
+        <Row>
+            <Col>
+                <div className="docslinks-title">Examples</div>
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx]: Do not hardcode strings.

File: ota-plus-web/app/reactapp-new/package.json [hunk 3]
```diff
@@ -5,6 +5,7 @@
   "main": "index.js",
   "scripts": {
     "start": "./node_modules/webpack/bin/webpack.js -w",
+    "dev": "webpack-dev-server --mode development --open --hot",
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp-new/package.json]: This is not needed in this ticket.

File: ota-plus-web/app/reactapp-new/src/index.tsx [hunk 4]
```diff
@@ -4,6 +4,6 @@ import React from 'react';
 import App from './App';
 
 ReactDOM.render(
-  <App />, 
-  document.getElementById('app')
+  <App />,
+  document.getElementById('root')
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp-new/src/index.tsx]: Why this has been changed? This is not part of this ticket.

File: ota-plus-web/app/reactapp-new/webpack.config.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
 const { resolve } = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp-new/webpack.config.js]: This is not a part of this ticket. Remove it please.

File: ota-plus-web/app/reactapp-new/webpack.config.js [hunk 6]
```diff
@@ -18,6 +19,10 @@ module.exports = {
     extensions: ['.ts', '.tsx', '.js'],
   },
   devtool: 'source-map',
+  devServer: {
+    contentBase: resolve(__dirname, 'src'),
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp-new/webpack.config.js]: This is not a part of this ticket. Remove it please.

File: ota-plus-web/app/reactapp-new/webpack.config.js [hunk 7]
```diff
@@ -65,4 +57,9 @@ module.exports = {
       },
     ],
   },
+  plugins: [
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp-new/webpack.config.js]: This is not a part of this ticket. Remove it please.

File: ota-plus-web/app/reactapp-new/package.json (hunk 3 above)
Comment (mhls) [reply ota-plus-web/app/reactapp-new/package.json]: its from Lukasz commit, i have made it on top of his change to have server setup done

File: ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx (hunk 2 above)
Comment (merltron) [reply ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx]: +1 Also want to handle the destination URLs for the doc links in L10 file (Izumi)

File: ota-plus-web/app/reactapp-new/package.json (hunk 3 above)
Comment (piotrcichonmee) [reply ota-plus-web/app/reactapp-new/package.json]: The core project is configures and this is not needed in this case.

File: ota-plus-web/app/assets/locales/en/translation-new.json [hunk 8]
```diff
@@ -1,3 +1,9 @@
 {
-    "common.test": "Hello new world"
+    "common.test": "Hello new world",
+    "docslinks.title": "Examples",
```
Comment (piotrcichonmee) [inline ota-plus-web/app/assets/locales/en/translation-new.json]: Use: dashboard.doclinks...

File: ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+import { TFunction } from 'i18next';
+
+import './DocsLinks.scss';
+
+interface IProps extends WithTranslation {
+    t: TFunction;
+}
+
+const DocsLinks: React.FC<IProps> = ({ t }) => {
+    return (
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx]: Use: const [t] = useTranslation();

File: ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+import './DocsLinks.scss';
+
+const DocsLinks: React.FC<{}> = () => {
+    const [t] = useTranslation();
+    return (
+        <Row>
+            <Col>
+                <div className="docslinks-title">{t('dashboard.docslinks.title')}</div>
+                <div className="docslinks-description">{t('dashboard.docslinks.description')}</div>
+                <ul  className="docslinks-links">
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx]: Unnecessary 2 spaces after ul.

File: ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+                    <li><a href="#">{t('dashboard.docslinks.links.devicegroup')}</a></li>
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx]: Use rel="noopener noreferrer" and target="_blank"

File: ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+                    <li><a href="#">{t('dashboard.docslinks.links.updateconfiguration')}</a></li>
+                    <li><a href="#">{t('dashboard.docslinks.links.campaigns')}</a></li>
+                    <li><a href="#">{t('dashboard.docslinks.links.troubleshoot')}</a></li>
+                </ul>
+            </Col>
+        </Row>
+    );
+};
+
+export default DocsLinks;
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx]: File should end with a new line

File: ota-plus-web/app/reactapp-new/webpack.config.js [hunk 13]
```diff
@@ -31,24 +31,11 @@ module.exports = {
         test: /\.scss$/,
         use: [
           {
-            loader: 'file-loader',
-            options: {
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp-new/webpack.config.js]: Why some many things have been removed?

Comment (mhls) [reply ota-plus-web/app/reactapp-new/webpack.config.js]: file-loader into style-loader -> the part is configuration of file loader.

File: ota-plus-web/app/reactapp-new/package.json [hunk 14]
```diff
@@ -63,6 +63,7 @@
     "antd": "^3.23.1",
     "i18next": "^17.0.12",
     "i18next-browser-languagedetector": "^3.0.3",
+    "i18next-xhr-backend": "^3.1.2",
```
Comment (laniszew) [inline ota-plus-web/app/reactapp-new/package.json]: Is this needed?

File: ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+  lineHeight: '17px'
+}));
+
+const List = styled.ul((_: Props) => ({
+  listStyle: 'none',
+  margin: 0,
+  padding: 0
+}));
+
+const DocsLinks: React.FC<{}> = () => {
```
Comment (laniszew) [inline ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx]: Don't use React.FC<> to type functional components, it breaks default props and other static properties

File: ota-plus-web/app/assets/locales/en/translation-new.json [hunk 16]
```diff
@@ -8,5 +8,11 @@
     "footer.links.cookie-policy.url": "",
     "footer.links.contact-us": "Contact us",
     "footer.links.contact-us.url": "",
-    "footer.copyright": "@ 2019 HERE"
+    "footer.copyright": "@ 2019 HERE",
+    "dashboard.docslinks.title": "Examples",
```
Comment (piotrsmol) [inline ota-plus-web/app/assets/locales/en/translation-new.json]: Please keep alphabetical order 😃

File: ota-plus-web/app/reactapp-new/src/components/common/Icon.tsx [hunk 17]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import styled from 'styled-components';
+
+enum Theme {
+  light = 'light',
+  dark = 'dark'
+}
+
+type IProps = {
```
Comment (piotrsmol) [inline ota-plus-web/app/reactapp-new/src/components/common/Icon.tsx]: Why IProps? It's not an interface

File: ota-plus-web/app/reactapp-new/src/components/common/Icon.tsx [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+      devices: '/assets/img/newapp/icons/devices_dark.svg',
+      deviceGroup: '/assets/img/newapp/icons/device_group_dark.svg',
+      softwareVersion: '/assets/img/newapp/icons/software_versions_dark.svg',
+      softwareUpdates: '/assets/img/newapp/icons/software_updates_dark.svg',
+    },
+    [Theme.light]: {
+
+    }
+  };
+  const iconType = (types[theme][type]) ? types[theme][type] : '';
```
[the remark this answers is not in the document]
Comment (mhls) [reply ota-plus-web/app/reactapp-new/src/components/common/Icon.tsx]: done

File: ota-plus-web/app/reactapp-new/src/components/common/Icon.tsx [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+
+type IProps = {
+  className?: string
+  type: string,
+  alt?: string,
+  theme?: Theme
+};
+
+export const Icon = ({ className, type, alt = type, theme = Theme.dark }: IProps) => {
+  const types: { [key: string]: any } = {
```
[the remark this answers is not in the document]
Comment (mhls) [reply ota-plus-web/app/reactapp-new/src/components/common/Icon.tsx]: done

File: ota-plus-web/app/reactapp-new/src/components/common/Icon.tsx (hunk 17 above)
Comment (mhls) [reply ota-plus-web/app/reactapp-new/src/components/common/Icon.tsx]: done

File: ota-plus-web/app/reactapp-new/src/components/common/Icon.tsx [hunk 20]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import styled from 'styled-components';
```
[the remark this answers is not in the document]
Comment (mhls) [reply ota-plus-web/app/reactapp-new/src/components/common/Icon.tsx]: removed

File: ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+import { Container, Title, ExternalLink } from '../common';
+import styled, { DefaultTheme } from 'styled-components';
+
+import { Icon } from '../common/Icon';
+
+type Props = {
+  theme: DefaultTheme
+};
+
+const Description = styled.div(({ theme }: Props) => ({
```
[the remark this answers is not in the document]
Comment (mhls) [reply ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx]: changed

File: ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx [hunk 22]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { useTranslation, UseTranslationResponse } from 'react-i18next';
+import { Container, Title, ExternalLink } from '../common';
+import styled, { DefaultTheme } from 'styled-components';
+
+import { Icon } from '../common/Icon';
```
[the remark this answers is not in the document]
Comment (mhls) [reply ota-plus-web/app/reactapp-new/src/components/DocsLinks/DocsLinks.tsx]: done

File: ota-plus-web/app/assets/locales/en/translation-new.json [hunk 23]
```diff
@@ -1,5 +1,11 @@
 {
     "common.test": "Hello new world",
+    "dashboard.docslinks.title": "Examples",
+    "dashboard.docslinks.description": "Real-life examples to help you understand the following tasks:",
+    "dashboard.docslinks.links.devicegroup": "Create device groups",
+    "dashboard.docslinks.links.updateconfiguration": "Create update configurations",
+    "dashboard.docslinks.links.campaigns": "Create campaigns",
+    "dashboard.docslinks.links.troubleshoot": "Troubleshoot campaign"
```
[the remark this answers is not in the document]
Comment (mhls) [reply ota-plus-web/app/assets/locales/en/translation-new.json]: done

File: ota-plus-web/app/assets/locales/en/translation-new.json [hunk 24]
```diff
@@ -1,5 +1,11 @@
 {
     "common.test": "Hello new world",
+    "dashboard.docslinks.title": "Examples",
+    "dashboard.docslinks.description": "Real-life examples to help you understand the following tasks:",
```
[the remark this answers is not in the document]
Comment (mhls) [reply ota-plus-web/app/assets/locales/en/translation-new.json]: updated accordingly to new layout

**Pull request merged**